### DISCUSSION TITLE
Connecting to private channel does not work!

### DISCUSSION THREAD

**Discussion by ezequidias** [qruto/laravel-wave#22] [Q&A] [answered]
The new laravel wave js improvements work perfectly! thank you very much for the support! [#Pull 10 Laravel Wave JS](https://github.com/qruto/laravel-wave-js/pull/10) @slavarazum 

When Broadcast::auth() is used, the logged in user will always be null!! this happens because the bearer token is not sent in the route header */wave*.

As it is not allowed to send header in EventSource , there are other possibilities to send the token!

one of them is to send in the url itself!! /wave?token= .....

I've tried using withCredentials but it doesn't work either!

I sent in some pull requests for a review!

#21 Add middleware to header auth (Laravel Wave)
[#11 ](https://github.com/qruto/laravel-wave-js/pull/11) Sending token in url (Laravel Wave JS)

**Comment by slavarazum** [comment, marked as answer]
Looks like `EventSource`, `WebSocket` and `XmlHttpRequest` APIs are [considered as legacy](https://github.com/whatwg/html/issues/2177#issuecomment-267270198).

`fetch` is our new hero.

Starting migrate to [@microsoft/fetch-event-source](https://github.com/Azure/fetch-event-source). So, all [Request options](https://developer.mozilla.org/en-US/docs/Web/API/Request/Request) will be available.

**Reply by ezequidias** [reply]
Fetch without a doubt is a much better solution!! ```fetch``` is our new hero.

You can disregard my pull requests!!

Thank you very much!!

```WaveConnector```
```WaveConnectorFetch```

**Reply by lianmaymesi** [reply]
There is no `WaveConnectorFetch` 

does not provide an export named 'WaveConnectorFetch'.

**Reply by slavarazum** [reply]
Do you need some extra abstraction over fetch?
